File: Data Structure II/stack/stack.go
package main

import (
	"fmt"
)

type Stack struct {
	items []int
}

func (s *Stack) Push(item int) {
	s.items = append(s.items, item)
}

func (s *Stack) Pop() int {
	if len(s.items) == 0 {
		fmt.Println("Stack is empty.")
		return -1
	}
	popped := s.items[len(s.items)-1]
	s.items = s.items[:len(s.items)-1]
	return popped
}

func (s *Stack) Display() {
	if len(s.items) == 0 {
		fmt.Println("Stack is empty.")
	} else {
		fmt.Println("Stack:", s.items)
	}
}

func (s *Stack) Contains(item int) bool {
	for _, v := range s.items {
		if v == item {
			return true
		}
	}
	return false
}

func main() {
	s := Stack{}
	var choice, num int

	for {
		fmt.Println("\nStack Menu:")
		fmt.Println("1. Push")
		fmt.Println("2. Pop")
		fmt.Println("3. Display Stack")
		fmt.Println("4. Check if Number Exists")
		fmt.Println("5. Exit")
		fmt.Print("Enter choice: ")
		fmt.Scan(&choice)

		switch choice {
		case 1:
			fmt.Print("Enter number to push: ")
			fmt.Scan(&num)
			s.Push(num)
		case 2:
			fmt.Println("Popped:", s.Pop())
		case 3:
			s.Display()
		case 4:
			fmt.Print("Enter number to check: ")
			fmt.Scan(&num)
			if s.Contains(num) {
				fmt.Println("Exists")
			} else {
				fmt.Println("Does not exist")
			}
		case 5:
			return
		default:
			fmt.Println("Invalid choice, try again.")
		}
	}
}

File: Data Structure II/stack/stack.rb
class Stack
  def initialize
    @stack = []
  end

  def push(item)
    @stack.push(item)
  end

  def pop
    return "Stack is empty." if @stack.empty?
    @stack.pop
  end

  def display
    return "Stack is empty." if @stack.empty?
    @stack
  end

  def contains?(item)
    @stack.include?(item)
  end
end

def menu
  s = Stack.new
  loop do
    puts "\nStack Menu:"
    puts "1. Push"
    puts "2. Pop"
    puts "3. Display Stack"
    puts "4. Check if Number Exists"
    puts "5. Exit"
    print "Enter choice: "
    choice = gets.chomp.to_i

    case choice
    when 1
      print "Enter number to push: "
      num = gets.chomp.to_i
      s.push(num)
    when 2
      puts "Popped: #{s.pop}"
    when 3
      puts "Stack: #{s.display}"
    when 4
      print "Enter number to check: "
      num = gets.chomp.to_i
      puts s.contains?(num) ? "Exists" : "Does not exist"
    when 5
      break
    else
      puts "Invalid choice, try again."
    end
  end
end

menu
